### PR TITLE
Enable checkstyle `failOnViolation`

### PR DESCRIPTION
**Pull request by timja**
Context https://github.com/jenkinsci/prism-api-plugin/pull/41#issuecomment-1144697022

### CODE REVIEW

**Comment by timja** [comment]
Something isn't right, I still can't get it to fail locally at commit https://github.com/jenkinsci/prism-api-plugin/pull/41/commits/66dcb92a0ef09d7bb84de42c7f7c62cd965b0b2b

File: pom.xml [hunk 1]
```diff
@@ -331,12 +331,14 @@
             <goals>
               <goal>checkstyle</goal>
             </goals>
-            <phase>verify</phase>
+            <phase>validate</phase>
```
Comment (timja) [inline pom.xml:334]: not sure whether to keep here or not, but waiting for unit test suite for linting is not great

File: pom.xml [hunk 2]
```diff
@@ -331,12 +331,14 @@
...
           </execution>
         </executions>
         <configuration>
           <linkXRef>false</linkXRef>
-          <failOnViolation>false</failOnViolation>
+          <violationSeverity>warning</violationSeverity>
+          <failOnViolation>true</failOnViolation>
+          <consoleOutput>true</consoleOutput>
```
Comment (timja) [inline pom.xml:341]: this makes the errors go to build log

File: pom.xml [hunk 3]
```diff
@@ -331,12 +331,14 @@
...
             </goals>
-            <phase>verify</phase>
+            <phase>validate</phase>
           </execution>
         </executions>
         <configuration>
           <linkXRef>false</linkXRef>
-          <failOnViolation>false</failOnViolation>
+          <violationSeverity>warning</violationSeverity>
+          <failOnViolation>true</failOnViolation>
```
Comment (timja) [inline pom.xml:340]: this should be the default and can be removed once working

File: pom.xml [hunk 4]
```diff
@@ -331,12 +331,14 @@
...
               <goal>checkstyle</goal>
             </goals>
-            <phase>verify</phase>
+            <phase>validate</phase>
           </execution>
         </executions>
         <configuration>
           <linkXRef>false</linkXRef>
-          <failOnViolation>false</failOnViolation>
+          <violationSeverity>warning</violationSeverity>
```
Comment (timja) [inline pom.xml:339]: this _should_  be causing the build to fail at https://github.com/jenkinsci/prism-api-plugin/pull/41/commits/66dcb92a0ef09d7bb84de42c7f7c62cd965b0b2b but it doesn't seem to be 😢

**Comment by uhafner** [comment]
Replaced by #907

**Pull request closed without merging**